src/movement_controls.py

```diff
@@ -0,0 +1,170 @@
+    current_velocity_y: float
+    current_velocity_turn: float
+    current_acceleration: float
+
+    def __init__(self, sprite: arcade.Sprite):
+        self.shadow_sprite = sprite
+        self.shadow_sprite.center_x = sprite.center_x
+        self.shadow_sprite.center_y = sprite.center_y
+        self.shadow_sprite.angle = sprite.angle
+        self.debug_world_boundary = DEFAULT_WORLD_SIZE
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_velocity_turn = 0
+        self.current_acceleration = 0
+        self.external_velocity_turn = 0
+
+        self.acceleration_rate = DEFAULT_ACCELERATION_RATE
+        self.brake_rate = DEFAULT_BRAKE_RATE
+        self.friction = DEFAULT_FRICTION
+        self.drive_speed = DEFAULT_DRIVE_SPEED
+        self.turn_speed = DEFAULT_TURN_SPEED
+        self.impact_buffer = 0
+
+    #   Drive input is called by the player and passed that player's input
+    #   to set what the vehicle's velocity and rotation should be
+    #   Does not change the vehicles position
+    def drive_input(
+        self, delta_time: float, input: PlayerInput, vehicle: arcade.Sprite
+    ):
+
+        if self.impact_buffer > 0:
+            self.impact_buffer -= 1
+            if self.impact_buffer <= 0:
+                self.external_velocity_turn = 0
+
+        # updates the vehicles intended  velocity and rotation based on type
+        acceleration_change = 0
+        self.shadow_sprite.center_x = vehicle.center_x
+        self.shadow_sprite.center_y = vehicle.center_y
+
+        if input.accelerate_axis.value:
+            if self.current_acceleration < 0:
+                acceleration_change += self.brake_rate
+            acceleration_change += self.acceleration_rate
+        elif input.brake_axis.value:
+            acceleration_change -= self.brake_rate
+        else:
+            if abs(self.current_acceleration) != 0:
+                if abs(self.current_acceleration) <= delta_time * self.friction:
+                    self.current_acceleration = 0
+                else:
+                    acceleration_change -= (
+                        math.copysign(1, self.current_acceleration) * self.friction
+                    )
+        # the current acceleration is used to have speed progress from zero to its maximum
+        self.current_acceleration = self.clamp(
+            self.current_acceleration + (acceleration_change * delta_time), -1, 1
+        )
+
+        car_angle = math.radians(vehicle.angle)
+
+        # find the X and Y movement based on the angle of the sprite
+        self.current_velocity_x = (
+            self.drive_speed
+            * math.cos(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+        self.current_velocity_y = (
+            self.drive_speed
+            * math.sin(car_angle)
+            * self.current_acceleration
+            * delta_time
+        )
+
+        # a stationary car cant turn
```

Comment:
It's not obvious to me how the logic of this comment is enforced in the line below.  Is there something missing?